zerver/tests/test_import_export.py

```diff
@@ -480,91 +480,89 @@ def test_import_realm(self) -> None:
 
+        def assert_realm_values(f):
+            orig_realm_result = f(original_realm)
+            imported_realm_result = f(imported_realm)
+            self.assertEqual(orig_realm_result, imported_realm_result)
+
         # test users
-        emails = [
-            {user.email for user in realm.get_admin_users()}
-            for realm in realms]
-        self.assertEqual(emails[0], emails[1])
-        emails = [
-            {user.email for user in realm.get_active_users().filter(is_bot=False)}
-            for realm in realms]
-        self.assertEqual(emails[0], emails[1])
+        assert_realm_values(
+            lambda r: {user.email for user in r.get_admin_users()}
+        )
+
+        assert_realm_values(
+            lambda r: {user.email for user in r.get_active_users()}
+        )
 
         # test stream
-        stream_names = [
-            {stream.name for stream in get_active_streams(realm)}
-            for realm in realms]
-        self.assertEqual(stream_names[0], stream_names[1])
+        assert_realm_values(
+            lambda r: {stream.name for stream in get_active_streams(r)}
+        )
 
         # test recipients
-        stream_recipients = [
-            get_stream_recipient(Stream.objects.get(name='Verona', realm=realm).id)
-            for realm in realms]
-        self.assertEqual(stream_recipients[0].type, stream_recipients[1].type)
+        def get_recipient_stream(r):
+            return get_stream_recipient(
+                Stream.objects.get(name='Verona', realm=r).id
+            )
+
+        def get_recipient_user(r):
+            return get_personal_recipient(
+                UserProfile.objects.get(full_name='Iago', realm=r).id
+            )
 
-        user_recipients = [
-            get_personal_recipient(UserProfile.objects.get(full_name='Iago', realm=realm).id)
-            for realm in realms]
-        self.assertEqual(user_recipients[0].type, user_recipients[1].type)
+        assert_realm_values(lambda r: get_recipient_stream(r).type)
+        assert_realm_values(lambda r: get_recipient_user(r).type)
 
         # test subscription
-        stream_subscription = [
-            Subscription.objects.filter(recipient=recipient)
-            for recipient in stream_recipients]
-        subscription_stream_user = [
-            {sub.user_profile.email for sub in subscription}
-            for subscription in stream_subscription]
-        self.assertEqual(subscription_stream_user[0], subscription_stream_user[1])
-
-        user_subscription = [
-            Subscription.objects.filter(recipient=recipient)
-            for recipient in user_recipients]
-        subscription_user = [
-            {sub.user_profile.email for sub in subscription}
-            for subscription in user_subscription]
-        self.assertEqual(subscription_user[0], subscription_user[1])
+        def get_users(r):
+            recipients = [get_recipient_stream(r), get_recipient_user(r)]
+            users = []
```

Comment:
You probably want `users` to be a `set()` here, and you can do `users.add()` in  the loop to avoid ordering flakes.

Reply:
This shouldn't cause a test flake as recipients is defined in a particular order `recipients = [get_recipient_stream(r), get_recipient_user(r)]` and we append the set of users for both of these to (`users.append({sub.user_profile.email for sub in subscriptions})`). Hence it is actually a List of a set and the list only contains two elements.